classes/navigationMenu/NavigationMenuDAO.inc.php

```diff
@@ -0,0 +1,303 @@
+		);
+
+		$returner = null;
+		if ($result->RecordCount() != 0) {
+			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Retrieve a navigation menu by title
+	 * @param $contextId int Context Id
+	 * @param $title string
+	 * @return NavigationMenu
+	 */
+	function getByTitle($contextId, $title) {
+		$params = array((int) $contextId);
+		$params[] = $title;
+		$result = $this->retrieve(
+			'SELECT * FROM navigation_menus WHERE context_id = ? and title = ?',
+			$params
+		);
+
+		$returner = null;
+		if ($result->RecordCount() != 0) {
+			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Check if a navigationMenu exists with the given title.
+	 * @param $title int
+	 * @param $contextId int
+	 * @return boolean
+	 */
+	function navigationMenuExistsByTitle($title, $contextId) {
+		$result = $this->retrieve(
+			'SELECT COUNT(*)
+			FROM	navigation_menus
+			WHERE	title = ? AND
+				context_id = ?',
+			array(
+				$title,
+				(int) $contextId
+			)
+		);
+		$returner = isset($result->fields[0]) && $result->fields[0] != 0 ? true : false;
+
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Get the locale field names.
+	 * @return array
+	 */
+	function getLocaleFieldNames() {
+	    return array();
+	}
+
+	/**
+	 * Internal function to return an NavigationMenu object from a row.
+	 * @param $row array
+	 * @return NavigationMenu
+	 */
+	function _fromRow($row) {
+		$navigationMenu = $this->newDataObject();
+		$navigationMenu->setId($row['navigation_menu_id']);
+		$navigationMenu->setTitle($row['title']);
+		$navigationMenu->setAreaName($row['area_name']);
+		$navigationMenu->setContextId($row['context_id']);
+		$navigationMenu->setDefault($row['is_default']);
+
+		return $navigationMenu;
+	}
+
+	/**
+	 * Insert a new NavigationMenu.
+	 * @param $navigationMenu NavigationMenu
+	 * @return int
+	 */
+	function insertObject($navigationMenu) {
+		$this->update(
+				'INSERT INTO navigation_menus
+				(title, area_name, context_id, is_default)
+				VALUES
+				(?, ?, ?, ?)',
+			array(
+				$navigationMenu->getTitle(),
+				$navigationMenu->getAreaName(),
+				(int) $navigationMenu->getContextId(),
+				(int) $navigationMenu->getDefault(),
+			)
+		);
+		$navigationMenu->setId($this->getInsertId());
+
+		return $navigationMenu->getId();
+	}
+
+	/**
+	 * Update an existing NavigationMenu
+	 * @param NavigationMenu $navigationMenu
+	 * @return boolean
+	 */
+	function updateObject($navigationMenu) {
+		$returner = $this->update(
+			'UPDATE	navigation_menus
+			SET	title = ?,
+				area_name = ?,
+				context_id = ?,
+				is_default = ?
+			WHERE	navigation_menu_id = ?',
+			array(
+				$navigationMenu->getTitle(),
+				$navigationMenu->getAreaName(),
+				(int) $navigationMenu->getContextId(),
+				(int) $navigationMenu->getDefault(),
+				(int) $navigationMenu->getId(),
+			)
+		);
+
+		return $returner;
```

Comment:
Just FYI, the return value of `DAO::update` is sometimes a cause of confusion. Devs sometimes expect it to return `true` when something was successfully found and updated, but that's not what `update` returns. The `return` value here is undocumented, so it might promote that kind of confusion :)

Suggest not bothering to return something unless it's documented and useful. (We're not particularly consistent about this in legacy code.)